strawberryfields/gbs/plot.py

```diff
@@ -0,0 +1,235 @@
+    """ Provides the coordinates for the graph nodes when given an input graph layout.
+
+    Args:
+        graph (nx.Graph): input graph
+        l (dict): dictionary of nodes and their respective coordinates
+
+    Returns:
+         Tuple: x and y coordinates for each node
+    """
+    n_x = []
+    n_y = []
+
+    for n in graph.nodes():
+        n_x.append(l[n][0])
+        n_y.append(l[n][1])
+
+    return {"x": n_x, "y": n_y}
+
+
+def edge_coords(graph: nx.Graph, l: dict) -> dict:
+    """ Provides the coordinates for the graph edges when given an input graph layout.
+
+        Args:
+            graph (nx.Graph): input graph
+            l (dict): dictionary of edges and their respective coordinates
+
+        Returns:
+             dict: x and y coordinates for beginning and end of each edge
+        """
+    e_x = []
+    e_y = []
+
+    for e in graph.edges():
+
+        start_x, start_y = l[e[0]]
+        end_x, end_y = l[e[1]]
+
+        e_x.append(start_x)
+        e_x.append(end_x)
+
+        e_y.append(start_y)
+        e_y.append(end_y)
+
+        e_x.append(None)
+        e_y.append(None)
+
+    return {"x": e_x, "y": e_y}
+
+
+web_green1 = '#84ba5b'
+web_green_light1 = '#e0edd6'
+web_green2 = '#3e9651'
+web_red1 = '#d35e60'
+web_red2 = '#cc2529'
+sf_cyan = '#3e9bb7'
+sf_cyan_light = '#6eb4c9'
+sf_lightgreen = '#daf0e9'
+sf_red = '#d35e60'
+sf_grey = '#efefef'
+light_grey = '#CDCDCD'
+
+graph_node_colour = web_green2
+graph_edge_colour = light_grey
+subgraph_node_colour = web_red2
+subgraph_edge_colour = web_red2
+
+graph_node_size = 14
+subgraph_node_size = 16
+
+
+def plot(graph: nx.Graph, subgraph: Optional[list] = None, size: float = 500) -> None:
+    """ Creates a plot.ly plot of the input graph.
+
+    The graph layout is fixed to be the Kamada-Kawai layout with an aspect ratio of 1:1. The
+    function can plot just the input graph or the graph with a specified subgraph highlighted.
+    The function uses the standard colour theme of green nodes, grey edges, and red highlighted
+    subgraph.
+
+    **Example usage**:
+        >>> graph = nx.complete_graph(10)
+        >>> fig = plot(graph, [0, 1, 2, 3])
+        >>> fig.show()
+
+        Args:
+            graph (nx.Graph): input graph
+            subgraph (list): list of nodes comprising the subgraph to highlight
+            size (dict): size of the plot
+
+        Returns:
+             plot.ly graph object
+        """
+
+    s = graph.subgraph(subgraph)
+    l = nx.kamada_kawai_layout(graph)
+
+    g_nodes = go.Scatter(
+        **_node_coords(graph, l),
+        mode='markers',
+        hoverinfo='text',
+        marker=dict(color=graph_node_colour, size=graph_node_size, line_width=2)
+    )
+
+    g_edges = go.Scatter(
+        **edge_coords(graph, l),
+        line=dict(width=1, color=graph_edge_colour),
+        hoverinfo='none',
+        mode='lines'
+    )
+
+    g_nodes.text = [str(i) for i in graph.nodes()]
```

Comment:
One thought --- does the gbs apps layer ever 'rename' or 're-index' the nodes, or are the original node names kept throughout the entire process?

Reply:
In fact, it might be nice to allow the user to (optionally) specify which node attributes show up in the tool tips, or even a list of node attributes to display.

Reply:
I believe the gbs apps layer maintains the original node names throughout the entire process. I believe that displaying node attributes can be customized in plotly, so your suggestion should be possible. I'd prefer to keep it in mind but not implement it for now

Reply:
@josh146 no we follow the input node labelling throughout (i.e., one could start with a graph that has nodes [1, 3, 65, 777]).

The only exception is when one takes raw samples from the device, as `sample.sample` returns samples generated from an adjacency matrix, but then `sample.subgraphs` gives subgraph samples that follow the node labelling of the input graph.